### PR TITLE
PROD-635 IA-3204 poll setMetadata result before calling startVM

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/PROD-635

Depends on https://github.com/broadinstitute/workbench-libs/pull/896

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
multi-test

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala [hunk 1]
```diff
@@ -108,7 +108,7 @@ class LeoPubsubMessageSubscriber[F[_]](
                     logger.error(ctx.loggingCtx, e)(s"Encountered an error for app ${ee.appId}, ${ee.getMessage}") >> handleKubernetesError(
                       ee
                     )
-                  case _ => F.unit
+                  case _ => logger.error(ctx.loggingCtx, ee)(s"Fail to process pubsub message")
```
Comment (kyuksel) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala]: Nit:
```suggestion
                  case _ => logger.error(ctx.loggingCtx, ee)(s"Failed to process pubsub message.")
```

**Review by kyuksel — APPROVED** [review]
👍🏼  with a nit inline. I also approved the accompanying `workbench-libs` PR so you could merge that and use its hash in this PR instead of the snap version.

**Comment by Qi77Qi** [comment]
jenkins retest

**Pull request merged**